### ISSUE TITLE
Verbeteren klikbaarheid/semantiek van File component

### ISSUE THREAD

**Issue by TessaViergever** [nl-design-system/mijn-services#556]
## Beschrijving

Volledige lijst met bevindingen (tokens, styling, structuur) staat in [het Den Haag issue ](https://github.com/nl-design-system/denhaag/issues/2026)zelf. Dit sub-issue focust specifiek op het accessibility (en semantiek) stukje.

## Taken

- [ ] Hele component geen klikbaar `<a>` meer maken, maar `<div>` met daarbinnen echte interactieve elementen (download-link, verwijder-button)
- [ ] Verwijder knop implementeren als echte `<button>` i.p.v. visueel nagebootst element
- [ ] Bestandsnaam + context (o.a. "Download") koppelen aan interactief element via `aria-describedby` of sr-only tekst

## Acceptatiecriteria

- [ ] Verwijder knop gedraagt zich als echte button (SR herkennen 'm los van de rest van het component)
- [ ] SR gebruiker hoort bestandsnaam + actie (bijv. "Download [bestandsnaam]") op het interactieve element zelf

## Resources

* [Slack bericht over File accessibility](https://codefornl.slack.com/archives/C01DAT4TRPF/p1775727449944719)
